crates/nu-protocol/src/errors/parse_error.rs

```diff
@@ -623,6 +624,13 @@ impl DidYouMean {
     pub fn new(possibilities_bytes: &[&[u8]], input_bytes: &[u8]) -> DidYouMean {
         DidYouMean(did_you_mean_impl(possibilities_bytes, input_bytes))
     }
+
+    pub fn look_for_suggestion(working_set: &StateWorkingSet, input_bytes: &[u8]) -> DidYouMean {
+        DidYouMean::new(&working_set.list_variables(), input_bytes)
+    }
```

Comment:
This captures some logic that we're using in the parser to actually run the suggestion logic.